### DISCUSSION TITLE
Class to apply a style to all elements inside a parent element

### DISCUSSION THREAD

**Discussion by verbose117** [tailwindlabs/tailwindcss#9413] [Ideas]
# Apply class of a child to a parent element
This will save time if you are working with an element that needs a common style that needs to applied to all children of it and also allows having more consistency between styles. Also allowing in overriding this config for an element, if it has applied a different value for the specific property. Thus, there is less redundant code and faster development.

For example, I need to set the cursor of all elements inside the `nav` to pointer, I might need to re-apply to it all elements.

Before: 
```css
<nav>
  <a class="cursor-pointer ..."></a>
  <a class="cursor-pointer ..."></a>
</nav>
```

After:
```css
<nav class=":children-cursor-pointer">
  <a></a>
  <a></a>
</nav>
```

> The class can be any of your choice but the implementation of this functionality would be great 👍🏻 !

I am willing to open a PR or work together with @tailwindlabs to implement this feature.

**Comment by adamwathan** [comment]
Hey! We've [talked about this in the past](https://github.com/tailwindlabs/tailwindcss/pull/6086) but there are some complex problems to solve around the declaration order that have blocked it.

Since adding support for arbitrary variants though you can write your example like this in the current version of Tailwind:


```html
<nav class="[&>*]:cursor-pointer">
  <a>...</a>
  <a>...</a>
</nav>
```

**Reply by verbose117** [reply]
Thanks 👍🏻 !
